### PR TITLE
Feat : Added populate.py to populate dummy data for the db

### PR DESCRIPTION
**Pull request by Priyansh61**
Fixes Issue : https://github.com/BitByte-TPC/alumni/issues/132

### Description of PR : 
PR adds `python3 manage.py populate` command to the application, this way the user can add all the dummy data in the server using a single command.

Avoided use of `python3 manage.py shell > add_xy.py` since this was causing some import statements conflict. And this way we DJango Command Manager to efficiently handle the populate command without any issue.

### CODE REVIEW

**Review by garg3133 — COMMENTED** [review]
Good work! A few points apart from reviews below:
* I think the command should be moved to the main app (`AlumniConnect`) as the command is not related to any specific application but a global command.
* It was throwing error on the first run and needed `defusedxml` package to be updated. Please do `pip freeze > requirements.txt` to update all the required dependencies.

File: applications/adminportal/management/commands/populate.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            self.stdout.write(self.style.ERROR("An error occured while adding data"))
+
+        try :
+            add_pass()
+            self.stdout.write(self.style.SUCCESS("Passwords added successfully"))
+        except :
+            self.stdout.write(self.style.ERROR("An error occured while adding passwords"))
+            # ... call other functions ...
+
+            self.stdout.write(self.style.SUCCESS("Population process completed successfully"))
```
[the remark this answers is not in the document]
Comment (Priyansh61) [reply applications/adminportal/management/commands/populate.py:40]: Refactored this part of code

File: scripts/add_batch.py [hunk 2]
```diff
@@ -1,5 +1,15 @@
+import os
+import django
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "AlumniConnect.settings.development")
+django.setup()
+
```
[the remark this answers is not in the document]
Comment (Priyansh61) [reply scripts/add_batch.py:5]: fixed

File: scripts/add_data.py [hunk 3]
```diff
@@ -1,17 +1,65 @@
...
 from django.contrib.auth.models import User
+from applications.alumniprofile.models import Profile, Batch
+from django.db import transaction
+
+def reg_no_gen(degree_, spec_, year):
+    degree = {"B.Tech": "1", "B.Des": '2', "M.Tech": '3', "M.Des": '4', "PhD": '5'}
+    spec = {"NA": '00', "CSE": "01", "ECE": "02", "ME": "03", "MT": "04", "NS": "05", "DS": "06"}
+    last_reg_no = Profile.objects.filter(year_of_admission=year).order_by('user__date_joined').last()
+    new_reg_no = (int(str(last_reg_no.reg_no)[-4:]) + 1) if last_reg_no else 1
+    return degree[degree_] + spec[spec_] + str(year)[2:] + str(convert_int(new_reg_no, 4))
```
[the remark this answers is not in the document]
Comment (Priyansh61) [reply scripts/add_data.py:11]: Done!

File: scripts/add_batch.py [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
 from applications.alumniprofile.models import Batch
 
-for num in range(2009, 2023):
-    b1 = Batch(batch = num)
-    b1.save()
\ No newline at end of file
+def add_batch() :
+    try :
+        batch_list = [Batch(batch = year) for year in range(2000, 2024)]
```
[the remark this answers is not in the document]
Comment (Priyansh61) [reply scripts/add_batch.py:10]: Done

File: scripts/add_data.py [hunk 5]
```diff
@@ -1,17 +1,65 @@
...
+    with transaction.atomic():
+        try:
+            for row in sheet.iter_rows(min_row=2, values_only=True):
+                roll_no, name, email, sex, dob, programme, branch, batch_year,year_of_admission = row
+
+                roll_no = str(int(roll_no))
+                email = str(email)
+
+                # User creation logic
+                user, user_created = User.objects.get_or_create(username=roll_no, defaults={'email': email})
```
[the remark this answers is not in the document]
Comment (Priyansh61) [reply scripts/add_data.py:32]: done

**Review by garg3133 — APPROVED** [review]
`xlrd` should also be removed from the requirement.txt, but otherwise looks good!

**Pull request merged**